File: source/math/Plane.java
package math;

import java.io.Serializable;

public class Plane implements Serializable {
    public Float3 normal;
    public Float3 point;

    public Plane(Float3 normal, Float3 point) {
        this.normal = normal;
        this.point = point;
    }

    public Plane(Plane plane) {
        normal = new Float3(plane.normal);
        point = new Float3(plane.point);
    }
}

File: source/math/Plane.kt
package math

import java.io.Serializable

class Plane : Serializable {
    var normal: Float3 = Float3()
    var point: Float3 = Float3()

    constructor()

    constructor(normal: Float3, point: Float3) {
        this.normal = normal
        this.point = point
    }

    constructor(plane: Plane) {
        normal = Float3(plane.normal)
        point = Float3(plane.point)
    }
}